Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+                case nameof(SdkFeature.Facebook):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.Facebook] = 1;
+                    break;
+                case nameof(SdkFeature.Google):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.Google] = 1;
+                    break;
+                case nameof(SdkFeature.Apple):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.Apple] = 1;
+                    break;
+                case nameof(SdkFeature.JWT):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.JWT] = 1;
+                    break;
+                case nameof(SdkFeature.ApiKey):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.ApiKey] = 1;
+                    break;
+                case nameof(SdkFeature.ServerApiKey):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.ServerApiKey] = 1;
+                    break;
+                case nameof(SdkFeature.Function):
+                    if (!IsCredentials(identifierNameSyntax, context))
+                    {
+                        break;
+                    }
+
+                    _realmFeaturesToAnalyse[SdkFeature.Function] = 1;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private bool IsCredentials(IdentifierNameSyntax identifierNameSyntax, GeneratorSyntaxContext context)
+        {
+            try
+            {
+                var parentType = context.SemanticModel.GetTypeInfo(identifierNameSyntax.Parent.ChildNodes().First()).Type.ToDisplayString();
+                return parentType == "Realms.Sync.Credentials" || parentType == "Realms.Sync.Credentials.AuthProvider";
+            }
+            catch (Exception ex)
+            {
+                Diagnostics.Add(SourceGenerator.Diagnostics.AnalyticsDebugInfo(
+                            $"{identifierNameSyntax} is likely some user defined syntax token.{Environment.NewLine}{ex.Message}"));
+                return false;
+            }
+        }
+
+        private string GetJsonPayload(bool pretty)
```

Comment:
We can benchmark if this is more or less efficient than calling string.format on a premade string (that we can generate manually)

Reply:
Do you still believe we have to look into this?